pkg/zfs/zfs_util.go

```diff
@@ -299,6 +299,11 @@ func buildVolumeResizeArgs(vol *apis.ZFSVolume) []string {
 		ZFSVolArg = append(ZFSVolArg, volsizeProperty)
 	}
 
+	if vol.Spec.ThinProvision == "no" {
+		reservationProperty := "reservation=" + vol.Spec.Capacity
+		ZFSVolArg = append(ZFSVolArg, "-o", reservationProperty)
+	}
```

Comment:
I believe after @cinapm's PR , we need to set reservation or refreservation based on the quota type? Also we need a change in line 295.

@cinapm Any thoughts here?